### PR TITLE
Working example in express

### PR DESCRIPTION
**Pull request by ryansobol**
I whipped up a working example rather than wait for someone else to do it. I still have a bunch of questions about Oz, Hawk, and Iron; It's been a while since college where I studied cryptography and security. But I wanted to get this PR out into the wild so others could have the opportunity to comment on where I might have fouled things up.

### CODE REVIEW

**Comment by hueniverse** [comment]
You should rename it `express.js` since it is specific to that framework (with hapi you just use the scarecrow plugin). Also need to fix the linting issues.

**Comment by ryansobol** [comment]
The file has been renamed and has been linted. :white_check_mark: 

Going into this, my goal was to demonstrate how to implement the Oz workflow in as little code and minimal dependencies as possible. I started with a simple Node `http` server, but quickly got annoyed with parsing methods and URIs manually. I considered using `hapi`, but settled on `express` as it's the most minimal of the two.

I hope you don't mind that I have a bunch of questions for you about this protocol. I'll start off with just one. :smile: 

When an Oz ticket is generated, it's given a `key` property that's a [random, 32-byte String](https://github.com/hueniverse/oz/blob/master/lib/ticket.js#L344-L349). For the life of me, I can't figure out how this secret key is used! It's not even really that secret as it's included, in plain text, within both the app ticket and user ticket JSON responses. Can you help me understand what's going on here?

**Comment by hueniverse** [comment]
The key is used by the underlying hawk layer for making authenticated requests. It's sent once over TLS to the client and then only used to generate a hash the server can validate.

**Comment by ryansobol** [comment]
Okay, I understand. 

But if Hawk's default algorithm for generating HMACs is sha256, why use a 32-byte String? Isn't the key just going to be padded with zeros until it reaches the algorithm's desired key length of 64 bytes?

**Comment by hueniverse** [comment]
I'm pretty sure it wants a 256bits key.

**Comment by ryansobol** [comment]
You know, that's what I thought too! Given the name sha**256** in all. Then I read this and got all confused.

> Block size:  the size of the data block the underlying hash algorithm operates upon. **For SHA-256, this is 512 bits**, for SHA-384 and SHA-512, this is 1024 bits.

https://tools.ietf.org/html/rfc4868#section-2.1

> If the length of the key is exactly the algorithm block size, use it as-is. If the key is shorter than the block size, lengthen it to exactly the block size by padding it on the right with zero bits.

https://tools.ietf.org/html/rfc4868#section-2.1.2

Honestly, I'm probably just misinterpreting something about this document. But I wanted to ask you because I figured you knew better.

My next question is this: When requesting an RSVP, what's the benefit of using the `key` property embedded in the App ticket vs the original `key` in the App credentials?

(BTW - Thank you for taking the time to answer these questions. You don't know how much I appreciate it. It's been a long time since I've looked this deeply into security protocols. It's fascinating!)

**Comment by hueniverse** [comment]
Those are different keys. Each ticket comes with its own key.

**Comment by ryansobol** [comment]
Right, but why have different keys? In other words, what's the security benefit?

**Comment by hueniverse** [comment]
Oz uses tickets. In order to get a ticket, we need to bootstrap the process and we're using the client Hawk credentials. But once we got those, we exchange them for a ticket which includes all the application state we need, scope, expiration, etc.

**Comment by HazCod** [comment]
A client example would be awesome too!

**Comment by ryansobol** [comment]
Hey @hueniverse, thanks for being a patient sport. Feel free to merge this pull request in, if you're satisfied. I'm going to play around with Oz some more and write up a dedicated Oz authentication web app for fun. I'll probably have more questions then.  :)

File: package.json [hunk 1]
```diff
@@ -23,7 +23,10 @@
   },
   "devDependencies": {
     "code": "2.x.x",
-    "lab": "7.x.x"
+    "lab": "7.x.x",
+    "body-parser": "1.x.x",
+    "express": "4.x.x",
+    "request": "2.x.x"
```
Comment (hueniverse) [inline package.json]: Not going to add deps for examples. The example should document what it needs.

**Comment by ycpatel813** [comment]
@ryansobo l may i know what is the final status of express example??

**Comment by ryansobol** [comment]
@ycpatel813 The current status of the express example is that it works. Sorry it's taken me so long to incorporate the feedback. I've separated the example's dependencies from Oz's dependencies. Also, I've separated the server code from the client code as well as demonstrated how to store the encryption password in an environment variable. Feedback is very welcome.

PS - Once your satisfied, @hueniverse, I'm happy to add a `client_with_connection_manager.js` example to this PR that demonstrates how to use the new `Oz.client.Connection()` API.

**Comment by hueniverse** [comment]
I am probably not going to get to this for a while. Just want to set expectations.

**Comment by hueniverse** [comment]
I think this needs to live somewhere else. I don't want to maintain it moving forward as it is time consuming and I don't use express. If you post this somewhere else, I'll link to the example from the readme.

**Pull request closed without merging**